tensorbay/client/dataset.py

```diff
@@ -545,22 +553,24 @@ def _upload_segment(
                 )
 
         if have_frame_id:
-            segment_filter = ((frame, None, False) for frame in segment)
+            segment_filter = ((frame, frame.frame_id, False) for frame in segment)
         else:
             segment_filter = (
-                (frame, 10 * index + 10, False) for index, frame in enumerate(segment)
+                (frame, from_timestamp(10 * index + 10), False)
+                for index, frame in enumerate(segment)
             )
 
         if skip_uploaded_files:
-            done_set = {
-                frame.frame_id.timestamp().timestamp: frame
-                for frame in segment_client.list_frames()
-            }
+            done_set = {frame.frame_id: frame for frame in segment_client.list_frames()}
```

Comment:
previously we use timestamp in frame_id to identify the same frame, but this version, the frame_id is used.
Does this work?